lib/muffin/base.rb

```diff
@@ -19,5 +19,29 @@ def initialize(request: nil, params: nil, scope: nil)
 
       assign_attributes
     end
+
+    private
+
+    def update_nested!(relation, entities)
+      entities = [entities] unless entities.is_a?(Array)
+
+      entities.each do |entity|
+        entity_attributes = entity.attributes.except(:_destroy)
+
+        if entity.respond_to?(:id) && entity.id.present?
+          model = relation.find(entity.id)
```

Comment:
here we can get a huge performance boost by fetching the relation into a hash id->model and then select from that hash instead of doing an sql query for each entity